gcp/src/main/java/org/apache/iceberg/gcp/gcs/GCSInputFile.java

```diff
@@ -24,14 +24,17 @@
 import org.apache.iceberg.gcp.GCPProperties;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.SeekableInputStream;
+import org.apache.iceberg.metrics.MetricsContext;
 
 public class GCSInputFile extends BaseGCSFile implements InputFile {
-  public static GCSInputFile fromLocation(String location, Storage storage, GCPProperties gcpProperties) {
-    return new GCSInputFile(storage, BlobId.fromGsUtilUri(location), gcpProperties);
+
+  public static GCSInputFile fromLocation(String location, Storage storage,
```

Comment:
Should this be public? Since it is a public constructor in a public class, we should deprecate the old constructor instead of directly removing it. If we want to avoid needing to deprecate in the future, then I think we should consider making this class and constructor package-private.

Reply:
Thanks, for the feedback. I have made the class and method package-private.